### ISSUE TITLE
Support individual dimension-tracking and new options interface

### ISSUE THREAD

**Issue by rob2d** [rob2d/use-viewport-sizes#9]
For the next release, will be providing a new interface which will be preferred over the previous parameter overloading. Example usage:

```
// default behavior
const [vpW, vpH] = useViewportSizes({ dimension: 'both' });
```

```
// with only one dimension observed, and `debounceTimeout` of `20` ms applied
const vpW = useViewportSizes({ dimension: 'w', debounceTimeout: 20 });
```

Input-option type:
```
    export type VPSizesHasher = (({ vpW: number, vpH: number }) => String);
    export type VPSizesOptions ={
        debounceTimeout?: number,
        throttleTimeout?: number,
        hasher?: VPSizesHasher,
        dimension?: 'w'|'h'|'both' = 'both'
    };

```

The interface will be backwards-compatible, but the default behavior for numbers as input will be to throttle eventually vs debounce as a more natural resize method.

Initially will just default to `debounceTimeout` behavior, but as a heads up to anyone using the library where it is going and why there may be a major semver version bump.

(input is welcome for anyone interested)

**Comment by rob2d** [comment]
Just about done -- will close once the documentation is complete and this can be published on main repo channel. For now, published on NPM under `v0.3.0-beta.0`.

Can see this working @ https://i8urr.csb.app/

**Comment by rob2d** [comment]
released w/ `v0.3.0` on NPM
